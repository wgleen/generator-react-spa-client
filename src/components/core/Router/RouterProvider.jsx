import React from 'react'
import PropTypes from 'prop-types'
import { BrowserRouter as Router } from 'react-router-dom'

const RouterProvider = (props) => {
  const { children } = props

  return <Router>{children}</Router>
}

RouterProvider.defaultProps = {
  children: null
}

RouterProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ])
}

export default RouterProvider
