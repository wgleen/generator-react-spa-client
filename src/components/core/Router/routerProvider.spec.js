import React from 'react'
import { render, screen } from '@testing-library/react'
import RouterProvider from './RouterProvider'

describe('<RouterProvider />', () => {
  it('should render correctly', () => {
    const component = render(<RouterProvider />)

    expect(component).toMatchSnapshot()

    screen.debug(component.container)

    component.unmount()
  })
})
