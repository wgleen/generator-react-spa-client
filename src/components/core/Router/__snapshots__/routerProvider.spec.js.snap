// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouterProvider /> should render correctly 1`] = `
<RouterProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    />
  </BrowserRouter>
</RouterProvider>
`;
